Mathlib/Analysis/Calculus/SecondDerivativeTest.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2024 Bjørn Kjos-Hanssen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bjørn Kjos-Hanssen
+-/
+import Mathlib.Analysis.Calculus.FirstDerivativeTest
+import Mathlib.Order.Interval.Set.Basic
+
+/-!
+# The Second-Derivative Test
+
+We prove the Second-Derivative test from calculus using the First-Derivative test.
+Source: [Wikipedia](https://en.wikipedia.org/wiki/Derivative_test#Proof_of_the_second-derivative_test).
+
+## Main results
+
+* `isLocalMin_of_deriv_deriv_pos`: The second-derivative test, minimum version.
+
+## Tags
+
+derivative test, calculus
+-/
+
+open Set Filter Topology
+
+
+/-- Insert a missing point between two adjacent open real intervals. -/
+theorem insert_Ioo₀ {x ε₀ ε₁ : ℝ} (hε₀ : ε₀ > 0) (hε₁ : ε₁ > 0):
+    insert x (Ioo (x - ε₀) x ∪ Ioo x (x + ε₁)) = Ioo (x - ε₀) (x + ε₁) :=
+  insert_Ioo ⟨by linarith,by linarith⟩
+
+
+section SecondDeriv
+
+variable {f : ℝ → ℝ} {x₀ : ℝ}
+
+/-- If the slope from a critical point `x₀` to `b > x₀` is positive then so is the derivative
+ at `b`. -/
+lemma slopeSimpPos {b : ℝ} (hb : x₀ < b) (hbf : 0 < slope (deriv f) x₀ b)
```

Comment:
```suggestion
lemma deriv_pos_of_lt_of_slope_pos {b : ℝ} (hb : x₀ < b) (hbf : 0 < slope (deriv f) x₀ b)
```
or similar. I think presumably every lemma in this PR has a dual for when the slope is negative, so at least you should ensure you leave room for both names.

Reply:
Thanks, I have tried to improve the names in that vein now.